decbinhex/src/main.nim

```diff
@@ -0,0 +1,81 @@
+# that is needed, otherwise it won't compile
+# Also gdExport pragma seems needed for signals to function properly
+  proc on_inbox_text_changed(intext: string) {.gdExport.} =
+    #print intext.len
+    if nonhex_chars in intext or intext.len == 0:
+      #Important! any proc called that also needs to deal with nodes will error if not called with self
+      #(or self as a parameter, but that looks odd since it's not+cannot be defined as an argument)
+      self.disable_all_buttons("Waiting for valid input.")
+      return
+    if intext.len > 63:
+      self.disable_all_buttons("Input too large, causes overflow!")
+      return
+    self.getNode("grid/a/outlabel").Label.text = "Waiting for button choice."
+    self.getNode("grid/hex_button").BaseButton.disabled = false
+    if intext.len > 15:
+      self.getNode("grid/hex_button").BaseButton.disabled = true
+      if hex_letters in intext:
+        self.getNode("grid/decimal_button").BaseButton.disabled = true
+        self.getNode("grid/binary_button").BaseButton.disabled = true
+        self.getNode("grid/a/outlabel").Label.text = "Input hexadecimal too large, causes overflow!"
+        return
+    if hex_letters in intext:
+      self.getNode("grid/decimal_button").BaseButton.disabled = true
+      self.getNode("grid/binary_button").BaseButton.disabled = true
+      self.getNode("grid/a/outlabel").Label.text = (
+       "Live hexadecimal conversion is:\n" & $intext.parseHexInt & " in decimal")
+      return
+    if intext.len < 20 and parseInt(intext) == 0:
+      self.disable_all_buttons("Waiting for valid input.")
+      return
+    if nonbinary_numbers in intext:
+      self.getNode("grid/binary_button").BaseButton.disabled = true
+      self.getNode("grid/decimal_button").BaseButton.disabled = false
+    else:
+      self.getNode("grid/binary_button").BaseButton.disabled = false
+      self.getNode("grid/decimal_button").BaseButton.disabled = false
+      if (self.getNode("grid/binary_check").BaseButton).is_pressed:
+        if intext.len < 20: self.getNode("grid/a/outlabel").Label.text = (
+         "Live binary conversion is:\n" & $intext.parseBinInt & " in decimal")
+        else: self.getNode("grid/a/outlabel").Label.text = ("Input decimal too large to parse for live convert, please use button")
+
+  proc submit_decimal() {.gdExport.} =
+    #called proc is imported from convert.nim
+    self.getNode("grid/a/outlabel").Label.text = convert_from_decimal(
+      self.getNode("grid/inbox").LineEdit.text)
+
+  proc submit_binary() {.gdExport.} =
+    self.getNode("grid/a/outlabel").Label.text = convert_from_binary(
+      self.getNode(
+      "grid/inbox").LineEdit.text)
+
+  proc submit_hex() {.gdExport.} =
+    self.getNode("grid/a/outlabel").Label.text = convert_from_hexadecimal(
+      self.getNode("grid/inbox").LineEdit.text)
+    #self.getNode("grid/inbox").queue_free()
```

Comment:
Why is this commented out?

Reply:
Just something I tried, not relevant to the project.

Reply:
Did you see that I removed the comment?